### PR TITLE
Implemented: Feature to update password when receive requirePasswordChange with login API(#3bdgrp)

### LINKED ISSUE #84: Implement feature to update password when receive requirePasswordChange with login API
## What is the motivation for adding/enhancing this feature?

The feature is specifically useful when the default password is set for the new/old user and the user needs to be obliged to change the password on login.

## What are the acceptance criteria?

- [ ] ...

## Can you complete this feature request by yourself?

- [ ] YES
- [ ] NO

### PR DESCRIPTION
**Pull request by shashwatbangar**
### Related Issues

Closes #84 
https://github.com/hotwax/preorder/commit/47c54de6f56b84b0a4ca7c1c8b479d1568e37961

### Short Description and Why It's Useful

Suppose user login for first time, we give him a default password, then he/she needs to set his own password so account can stay secure.

### Screenshots of Visual Changes before/after (If There Are Any)

**IMPORTANT NOTICE** - Remember to update `CHANGELOG.md` with description of your change

### Contribution and Currently Important Rules Acceptance

- [x] I read and followed [contribution rules](https://github.com/DivanteLtd/vsf-capybara/blob/master/CONTRIBUTING.md)

### CODE REVIEW

File: src/views/update-password.vue [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+          showToast(this.$t('Something went wrong'));
+          console.error("error", resp.data._ERROR_MESSAGE_);
+        }
+      } catch (err) {
+        this.showPassword = false
+        this.currentPassword =''
+        this.newPassword =''
+        this.newPasswordVerify =''
+        resetForm({
+            ...form,
```
Comment (rathoreprashant) [inline src/views/update-password.vue:173]: indentation issue

File: src/views/login.vue [hunk 2]
```diff
@@ -72,11 +72,15 @@ export default defineComponent({
...
+        if(data.requirePasswordChange && data.token) {
+          this.username = ''
+          this.password = ''
+          this.$router.push({ name: 'update-password', params: { username , token: data.token}})
+        } else if (data.token) {
           this.username = ''
           this.password = ''
           this.$router.push('/')
-        }
+        }
```
Comment (rathoreprashant) [inline src/views/login.vue:83]: unwanted indentation

File: src/views/update-password.vue (hunk 1 above)
Comment (shashwatbangar) [reply src/views/update-password.vue:173]: FIXED IT SIR.

File: src/views/login.vue (hunk 2 above)
Comment (shashwatbangar) [reply src/views/login.vue:83]: Fixed it.

File: src/services/UserService.ts [hunk 3]
```diff
@@ -31,10 +31,19 @@ const setUserTimeZone = async (payload: any): Promise <any>  => {
     data: payload
   });
 }
+const updatePassword = async (payload: any): Promise <any>  => {
+  return client({
+    url: "updatePassword",
+    method: "post",
+    baseURL: process.env.VUE_APP_BASE_URL,
```
Comment (rathoreprashant) [inline src/services/UserService.ts:38]: baseURL should be handled if .env has not VUE_APP_BASE_URL

File: src/store/modules/user/actions.ts [hunk 4]
```diff
@@ -18,10 +18,17 @@ const actions: ActionTree<UserState, RootState> = {
     try {
       const resp = await UserService.login(username, password)
       if (resp.status === 200 && resp.data) {
-        if (resp.data.token) {
-            commit(types.USER_TOKEN_CHANGED, { newToken: resp.data.token })
-            dispatch('getProfile')
-            return resp.data;
+        if (resp.data.requirePasswordChange && resp.data.token) {
+          showToast(translate('Your password expired. Please update your password.'));
```
Comment (disha1202) [inline src/store/modules/user/actions.ts:22]: Add static text in en.json file

**Review by disha1202 — CHANGES_REQUESTED** [review]

File: src/store/modules/user/actions.ts (hunk 4 above)
Comment (shashwatbangar) [reply src/store/modules/user/actions.ts:22]: Done ma'am!

**Comment by k2maan** [comment]
Closing. Will be done from scratch later.

**Pull request closed without merging**